### ISSUE TITLE
Pagination not working when using PrivateKey Authorization Mode

### ISSUE THREAD

**Issue by rwbenchmarkcapital** [okta/okta-sdk-dotnet#611]
### Describe the bug?

I noticed after upgrading to v6 of the SDK and making the necessary code changes, that automatic pagination was no longer working when, for example, I was trying to retrieve all applications. E.g. this code - 

`var allApps = await applicationApi.ListApplications(limit: 100).ToListAsync();`

What would happen is, it would seemingly receive the first page only, as the total count would always match whatever I set as the limit value.

I tried to switch to manual pagination, which is when I discovered the issue, thanks to Fiddler. I was able to see that the first call was successful, however for the subsequent call, it was returning a 401 Unauthorized error (which was being swallowed by the SDK). I could see in Fiddler that for this subsequent request, it had appended the full Bearer/Access Token value for a second time in the authorization header. E.g. -

`Bearer: [Access Token Value], Bearer [Access Token Value]`

I had a look at the Okta SDK code, and found what I believe to be the culprit in `OktaPagedCollectionEnumerator.MoveNextAsync` - 

```
if (Okta.Sdk.Client.Configuration.IsPrivateKeyMode(_configuration))
{
     var accessToken = await _oAuthTokenProvider.GetAccessTokenAsync(cancellationToken: _cancellationToken);
    _nextRequest.HeaderParameters.Add("Authorization", $"Bearer {accessToken}");
}
```

This code is adding a Bearer token to the headers collection, even if there is already one present. Hence the behaviour I've witnessed.

### What is expected to happen?

When using PrivateKey authorization mode, pagination over collections in the Okta SDK should work both via the automatic or manual methods.

### What is the actual behavior?

When using PrivateKey authorization mode, only the first page is ever returned - subsequent page requests are met with an unauthorized error response.

### Reproduction Steps?

Simply set the Okta Configuration to Private Key, and attempt to retrieve a collection, for example ListApplications(), where that collection would consist of multiple pages.

### Additional Information?

As a temporary work around, I have created a custom ApiClient implementation, that derives from the default one and overrides (via `new`) the GetAsync method so that I can remove one of the Authorization header entries if multiple are present - 

```
    public class FixedOktaApiClient : ApiClient, IAsynchronousClient
    {
        public FixedOktaApiClient(string oktaDomain)
            : base(oktaDomain)
        {
        }

        public new Task<ApiResponse<T>> GetAsync<T>(string path, RequestOptions options, IReadableConfiguration configuration = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken))
        {
            var authHeaders = options.HeaderParameters["Authorization"];
            if (authHeaders.Count > 1)
            {
                authHeaders.RemoveAt(1);
            }

            return base.GetAsync<T>(path, options, configuration, cancellationToken);
        }
    }
```

It's obviously not great, there's probably a better way to do it, but it works for the time-being!

### .NET Version

6.0.306

### SDK Version

6.0.306

### OS version

BuildNumber  Caption                          OSArchitecture  Version
19044        Microsoft Windows 10 Enterprise  64-bit          10.0.19044

**Comment by laura-rodriguez** [comment]
Hi @rwbenchmarkcapital,

Thanks for reporting this issue. I'll file an internal ticket for our team to review and prioritize.

Internal Ref: OKTA-563357

**Comment by laura-rodriguez** [comment]
Fixed in 6.0.2
